client/modules/IDE/components/Editor/index.jsx

```diff
@@ -141,546 +228,443 @@ class Editor extends React.Component {
+      'autoCloseBrackets',
+      mergedState.autocloseBracketsQuotes
+    );
+    cm.current.setOption('lineNumbers', mergedState.lineNumbers);
+    cm.current.setOption('lineWrapping', mergedState.linewrap);
+  }, [
+    mergedState.theme,
+    mergedState.autocloseBracketsQuotes,
+    mergedState.lineNumbers,
+    mergedState.linewrap
+  ]);
+
+  useEffect(() => {
+    delete cm.current.options.lint.options.errors;
 
     const replaceCommand =
       metaKey === 'Ctrl' ? `${metaKey}-H` : `${metaKey}-Option-F`;
-    this._cm.setOption('extraKeys', {
-      Tab: (cm) => {
-        if (!cm.execCommand('emmetExpandAbbreviation')) return;
+    cm.current.setOption('extraKeys', {
+      Tab: (_cm) => {
+        if (!_cm.execCommand('emmetExpandAbbreviation')) return;
         // might need to specify and indent more?
-        const selection = cm.doc.getSelection();
+        const selection = _cm.doc.getSelection();
         if (selection.length > 0) {
-          cm.execCommand('indentMore');
+          _cm.execCommand('indentMore');
         } else {
-          cm.replaceSelection(' '.repeat(INDENTATION_AMOUNT));
+          _cm.replaceSelection(' '.repeat(INDENTATION_AMOUNT));
         }
       },
       Enter: 'emmetInsertLineBreak',
       Esc: 'emmetResetAbbreviation',
       [`${metaKey}-Enter`]: () => null,
       [`Shift-${metaKey}-Enter`]: () => null,
       [`${metaKey}-F`]: 'findPersistent',
-      [`Shift-${metaKey}-F`]: this.tidyCode,
+      [`Shift-${metaKey}-F`]: tidyCode,
       [`${metaKey}-G`]: 'findPersistentNext',
       [`Shift-${metaKey}-G`]: 'findPersistentPrev',
       [replaceCommand]: 'replace',
       // Cassie Tarakajian: If you don't set a default color, then when you
       // choose a color, it deletes characters inline. This is a
       // hack to prevent that.
-      [`${metaKey}-K`]: (cm, event) =>
-        cm.state.colorpicker.popup_color_picker({ length: 0 }),
+      [`${metaKey}-K`]: (_cm, event) =>
+        _cm.state.colorpicker.popup_color_picker({ length: 0 }),
       [`${metaKey}-.`]: 'toggleComment' // Note: most adblockers use the shortcut ctrl+.
     });
 
-    this.initializeDocuments(this.props.files);
-    this._cm.swapDoc(this._docs[this.props.file.id]);
-
-    this._cm.on(
-      'change',
-      debounce(() => {
-        this.props.setUnsavedChanges(true);
-        this.props.hideRuntimeErrorWarning();
-        this.props.updateFileContent(this.props.file.id, this._cm.getValue());
-        if (this.props.autorefresh && this.props.isPlaying) {
-          this.props.clearConsole();
-          this.props.startSketch();
+    if (cm.current) {
+      const docs = {};
+      files.forEach((fileitem) => {
+        if (fileitem.name !== 'root') {
+          docs[fileitem.id] = CodeMirror.Doc(
+            fileitem.content,
+            getFileMode(fileitem.name)
+          ); // eslint-disable-line
         }
-      }, 1000)
-    );
+      });
+      cm.current.swapDoc(docs[file.id]);
+    }
```

Comment:
I think this needs to be in a separate effect because this effect does not have a dependency on `files`.